disco_aws_automation/disco_vpc.py

```diff
@@ -484,111 +286,103 @@ def _configure_environment(self):
 
             if vpc_cidr is None:
                 raise VPCConfigError('Cannot create VPC %s. No subnets available' % self.environment_name)
 
         # Create VPC
-        vpc_conn = VPCConnection()
-        self.vpc = vpc_conn.create_vpc(vpc_cidr)
-        keep_trying(300, self.vpc.add_tag, "Name", self.environment_name)
-        keep_trying(300, self.vpc.add_tag, "type", self.environment_type)
+        self.vpc = self.boto3_ec2.create_vpc(CidrBlock=str(vpc_cidr))['Vpc']
+        waiter = self.boto3_ec2.get_waiter('vpc_available')
+        waiter.wait(VpcIds=[self.vpc['VpcId']])
+        ec2 = boto3.resource('ec2')
+        vpc = ec2.Vpc(self.vpc['VpcId'])
+        tags = vpc.create_tags(Tags=[{'Key': 'Name', 'Value': self.environment_name},
+                                     {'Key': 'type', 'Value': self.environment_type}])
         logging.debug("vpc: %s", self.vpc)
+        logging.debug("vpc tags: %s", tags)
 
-        # Create metanetworks (subnets, route_tables and security groups)
-        self._networks = self._create_new_meta_networks()
-        for network in self.networks.values():
-            network.create()
-
-        dhcp_options = self._configure_dhcp()
-        self.vpc.connection.associate_dhcp_options(dhcp_options.id, self.vpc.id)
+        dhcp_options = self._configure_dhcp()[0]
+        self.boto3_ec2.associate_dhcp_options(DhcpOptionsId=dhcp_options['DhcpOptionsId'],
+                                           VpcId=self.vpc['VpcId'])
 
         # Enable DNS
-        vpc_conn.modify_vpc_attribute(self.vpc.id, enable_dns_support=True)
-        vpc_conn.modify_vpc_attribute(self.vpc.id, enable_dns_hostnames=True)
+        self.boto3_ec2.modify_vpc_attribute(
+            VpcId=self.vpc['VpcId'], EnableDnsSupport={'Value': True})
+        self.boto3_ec2.modify_vpc_attribute(
+            VpcId=self.vpc['VpcId'], EnableDnsHostnames={'Value': True})
 
-        # Configure security group rules
+        # Create metanetworks (subnets, route_tables and security groups)
+        self._networks = self._create_new_meta_networks()
         for network in self.networks.values():
-            self._add_sg_rules(network)
-
-        # Set up security group rules
-        self._open_customer_ports()
+            network.create()
 
-        # Allow ICMP (ping, traceroute & etc) and DNS traffic for all subnets
-        for network in self.networks.values():
-            self.vpc.connection.authorize_security_group(
-                group_id=network.security_group.id,
-                ip_protocol="icmp",
-                from_port=-1,
-                to_port=-1,
-                cidr_ip=vpc_cidr
-            )
-            self.vpc.connection.authorize_security_group(
-                group_id=network.security_group.id,
-                ip_protocol="udp",
-                from_port=53,
-                to_port=53,
-                cidr_ip=vpc_cidr
-            )
+        # Configure security group rules for all meta networks
```

Comment:
Yes, I didn't think of that. After making some changes, we can now use the same update methods during VPC creation.